libsolidity/interface/StandardCompiler.cpp

```diff
@@ -236,7 +236,11 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 		return formatFatalError("JSONError", "Only \"Solidity\" is supported as a language.");
 
 	Json::Value const& sources = _input["sources"];
-	if (!sources)
+
+	if (!sources.isObject() && !sources.isNull())
```

Comment:
Why do we need `isNull` here?

Is `IsObject` true when the value is null?

Reply:
nope.. but `null` is also not an object.. so if there is no `sources` object at all, it is also not an object.. `!sources.isObject()` is also `true` - that's why I do the additional check here.

however, just in case you didn't noticed: I introduced a slightly different error message here: `Sources input is not a JSON object.` in contrast to `Source input is not a JSON object.` - from my point of view its nice nice to distinguish between `sources` is not a json object, and if items of `sources` - so individual members of the `sources` object - `source` is not a json object.

Reply:
> so if there is no sources object at all, it is also not an object

Ah, yes that makes sense :)